p2p/client.go

```diff
@@ -504,8 +501,8 @@ func (c *Client) blockGossipReceived(ctx context.Context, block []byte) {
 	}
 	c.setLatestSeenHeight(gossipedBlock.Block.Header.Height)
 
-	// Received block is cached and  no longer needed to request using block-sync
-	c.heightToSkip[gossipedBlock.Block.Header.Height] = struct{}{}
+	// Received block is cached and no longer needed to request using block-sync
+	delete(c.heightToSkip, gossipedBlock.Block.Header.Height)
```

Comment:
are you sure this is right?

I can't understand this

```
// blockSyncReceived is called on reception of new block via block-sync protocol
func (c *Client) blockSyncReceived(block *P2PBlock) {
	err := c.localPubsubServer.PublishWithEvents(context.Background(), *block, map[string][]string{EventTypeKey: {EventNewBlockSyncBlock}})
	if err != nil {
		c.logger.Error("Publishing event.", "err", err)
	}
	// Received block is cached and  no longer needed to request using block-sync
	c.heightToSkip[block.Block.Header.Height] = struct{}{}
}

// blockSyncReceived is called on reception of new block via gossip protocol
func (c *Client) blockGossipReceived(ctx context.Context, block []byte) {
	var gossipedBlock P2PBlock
	if err := gossipedBlock.UnmarshalBinary(block); err != nil {
		c.logger.Error("Deserialize gossiped block", "error", err)
	}
	err := c.localPubsubServer.PublishWithEvents(context.Background(), gossipedBlock, map[string][]string{EventTypeKey: {EventNewGossipedBlock}})
	if err != nil {
		c.logger.Error("Publishing event.", "err", err)
	}
	err = c.AddBlock(ctx, gossipedBlock.Block.Header.Height, block)
	if err != nil {
		c.logger.Error("Adding  block to blocksync store.", "err", err, "height", gossipedBlock.Block.Header.Height)
	}
	c.setLatestSeenHeight(gossipedBlock.Block.Header.Height)

	// Received block is cached and no longer needed to request using block-sync
	delete(c.heightToSkip, gossipedBlock.Block.Header.Height)
}
```

Please dcoument heightToSkip field

Reply:
aka heightToSkipWhat

Reply:
definely wrong code. fixed. 
i renamed heightToSkip to blockAlreadyReceived. basically it is used to keep track of blocks received that are not required to request anymore.